src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php

```diff
@@ -63,21 +64,30 @@ public function hasCacheableSupportsMethod(): bool
     }
 
     /**
-     * Checks if the given class has any getter method.
+     * Checks if the given class has any getter or setter method.
      */
-    private function supports(string $class): bool
+    private function supports(string $class, array $context): bool
     {
         if (null !== $this->classDiscriminatorResolver && $this->classDiscriminatorResolver->getMappingForClass($class)) {
             return true;
         }
 
-        $class = new \ReflectionClass($class);
-        $methods = $class->getMethods(\ReflectionMethod::IS_PUBLIC);
-        foreach ($methods as $method) {
-            if ($this->isGetMethod($method)) {
-                return true;
+        $reflection = new \ReflectionClass($class);
+
+        // fallback to "read" to ensure backward compatibility
```

Comment:
Talking about BC while there is no deprecated code path feels weird. Do you plan to deprecate not setting that option explicitly in another PR on the dev branch? If not, I'd say better remove the comment or explain the reasoning otherwise if that fallback is going to be the default forever.